src/fmu/dataio/dataio.py

```diff
@@ -751,6 +751,15 @@ def _check_obj_if_file(self, obj: Any) -> Any:
 
         return obj
 
+    def _check_table_index(self):
+        """Removing table_index if None"""
+        try:
+            if self._metadata["data"]["table_index"] is None:
+                del self._metadata["data"]["table_index"]
+                logger.debug("Table index was none, removed")
+        except KeyError:
+            logger.debug("No table index in the first place")
```

Comment:
```
if not self._metadata["data"].get("table_index"):
    del self._metadata["data"]["table_index"]
    logger.debug("Table index was none, removed")
else:
     logger.debug("No table index in the first place")
```
If the class is table, should it be more than `logger.debug()` ? E.g. a `UserWarning`?

Reply:
> ```
> if not self._metadata["data"].get("table_index"):
>     del self._metadata["data"]["table_index"]
>     logger.debug("Table index was none, removed")
> else:
>      logger.debug("No table index in the first place")
> ```

Possibly redundant comment if everything is going away anyway (common removal of Nones later), but first line above will evaluate to True both if table_index is None and if it's not there. But the del operation will fail with KeyError if it is not there.

Suggest something like:
```
if "table_index" in self._metadata["data"] and self._metadata["data"]["table_index"] is None:
    del self._metadata["data"]["table_index"]
```

...if it is kept.

Reply:
My bad, just delete the "not" in the if... and it should work

Reply:
Now removed